### PR TITLE
Settings menu

### PR DESCRIPTION
**Pull request by ddbrown30**
I added the settings menu. This comes along with a bit of refactoring to make things more consistent and reliable but I tried not to touch too much.

For the moment, I just split the options into two tabs, world and user. All the world settings are in the world tab and all the user settings are in the user tab. The intention here is to eventually break these down into more tabs for better organization but I'll leave that to you to decide on and implement.

The biggest change in functionality is the user settings. Anything that was a client setting is now a user setting. User settings are saved to the flags on the user. This ensures that the setting travels with the user to different devices and is not wiped by clearing the cache.

The menu probably needs some CSS love but I'll also leave that to you. It looks fine as is, but it could use some spacing and padding adjustments and it would be nice if the apply button was pinned to the bottom of the window.

I tested every single option by hand. I'll never claim that there are no bugs but I feel pretty confident that all the core functionality is working (or at least isn't any more broken than it was before).

### CODE REVIEW

File: betterrolls-swade2/scripts/utils.js [hunk 1]
```diff
@@ -132,3 +134,122 @@ export async function set_or_update_condition(condition_id, actor) {
...
+    });
+  }
+
+  /**
+   * Register a single setting using the provided key and setting data
+   * @param {*} key 
+   * @param {*} metadata 
+   * @returns {ClientSettings.register}
+   */
+  static registerSetting(key, metadata) {
```
[the remark this answers is not in the document]
Comment (javierriveracastro) [reply betterrolls-swade2/scripts/utils.js:174]: This looks like unnecessary clutter to me.

**Comment by javierriveracastro** [comment]
This is more complicated, but there are some things that I don't like. Probably result of too many time using Python, but...

- I hate * imports. Sometimes you can need to edit the file in a text editor (not a fancy IDE) and you need to be able to know where things came from.
- No global scope variables (constants are OK). So that cascade count makes me nervous. I'll move it under game.swbr

It also has some merge conflicts with your other branchs, so I will merge this from my IDE that is way easier than github, so I close this here (it is still merged).

File: betterrolls-swade2/scripts/settings_config.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        return mergeObject(super.defaultOptions, {
+          id: 'brsw-settings-config',
+          classes: ['sheet'],
+          template: 'modules/betterrolls-swade2/templates/settings_config.html',
+          resizable: false,
+          minimizable: false,
+          title: game.i18n.localize('BRSW.Settings'),
+          width: 800,
+          height: 700,
+          resizable: true,
```
[the remark this answers is not in the document]
Comment (ddbrown30) [reply betterrolls-swade2/scripts/settings_config.js:19]: Oh, this is a good call out. It has resizable: false just above. Bad copy+paste. You already merged the change, though. Can you make the fix?

**Comment by ddbrown30** [comment]
> This is more complicated, but there are some things that I don't like. Probably result of too many time using Python, but...
> 
>     * I hate * imports. Sometimes you can need to edit the file in a text editor (not a fancy IDE) and you need to be able to know where things came from.
> 
>     * No global scope variables (constants are OK). So that cascade count makes me nervous. I'll move it under game.swbr
> 
> 
> It also has some merge conflicts with your other branchs, so I will merge this from my IDE that is way easier than github, so I close this here (it is still merged).

I can't speak to the imports thing. My JS knowledge is pretty newb level so I could definitely be doing stuff in a non-optimal way.

For the global variable, I 100% agree. I couldn't find a better place for it but I forgot about the game variable. That's a smart place to put it.

File: betterrolls-swade2/scripts/settings_config.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (javierriveracastro) [reply betterrolls-swade2/scripts/settings_config.js:19]: Sure.

**Pull request closed without merging**